modules/core/src/main/java/org/apache/ignite/internal/binary/BinaryMetadata.java

```diff
@@ -104,6 +118,13 @@ public BinaryMetadata(int typeId, String typeName, @Nullable Map<String, BinaryF
             schemaIds = Collections.emptySet();
 
         this.isEnum = isEnum;
+
+        if (enumMap != null) {
+            this.ordinalToNames = new LinkedHashMap<>(enumMap);
+            this.nameToOrdinal = new LinkedHashMap<>(enumMap.size());
+            for (Map.Entry<Integer, String> e: ordinalToNames.entrySet())
+                this.nameToOrdinal.put(e.getValue().toUpperCase(), e.getKey());
```

Comment:
I do not think we have to have case insensitive enum names. In java they are case sensitive, also this may cause a performance drop when we will do conversion from VARCHAR ot ENUM too often.